test/test_ops.py

```diff
@@ -171,6 +171,24 @@ def unsupported(dtype):
 
         self.assertEqual(supported_backward_dtypes, claimed_backward_supported, msg=msg)
 
+    # Validates that each OpInfo works correctly on different CUDA devices
+    @skipCUDAIfRocm
+    @onlyCUDA
+    @deviceCountAtLeast(2)
+    @ops(op_db, allowed_dtypes=(torch.float32, torch.long))
+    def test_multiple_devices(self, devices, dtype, op):
+        for cuda_device_str in devices:
+            cuda_device = torch.device(cuda_device_str)
+            # NOTE: only tests on first sample
+            samples = op.sample_inputs(cuda_device, dtype)
+            sample = samples[0]
+            result = op(sample.input, *sample.args, **sample.kwargs)
+
+            if isinstance(result, torch.Tensor):
+                self.assertTrue(result.device == cuda_device)
+            else:  # assumes that result is an iterable of tensors
```

Comment:
I think we have a function to test if a value is actually an iterable of tensors that we can use here.

I don't think every PyTorch op produces a tensor or iterable of tensors. torch.equal(), for example, produces a bool.

Reply:
`is_iterable_of_tensors` is here:

https://github.com/pytorch/pytorch/blob/9f3167ebdf85251428da525965b5006fc32b9e7e/torch/testing/_internal/common_utils.py#L720

There may be some improvements we can make to it but I think it will do what we need

Reply:
Thank you, Mike! `is_iterable_of_tensors` is now used.

Reply:
Thank you, Mike! is_iterable_of_tensors is now used.